src/openrct2/windows/ride_construction.c

```diff
@@ -539,14 +531,22 @@ static sint32 ride_get_alternative_type(rct_ride *ride)
 		ride->type;
 }
 
+/* move to ride.c */
+static void close_ride_window_for_construction(rct_windownumber number)
+{
+	rct_window* w = window_find_by_number(WC_RIDE, number);
+	if (w != NULL && w->page == 1)
```

Comment:
It might be because building can have a direct effect on the number of trains and cars per train. Too much interference.